internal/repository/root.go

```diff
@@ -23,10 +24,18 @@ func (r *Repository) InitializeRoot(ctx context.Context, signer sslibdsse.Signer
 		return err
 	}
 
-	rawKey := signer.Public()
-	publicKey, err := sslibsv.NewKey(rawKey)
-	if err != nil {
-		return err
+	var (
+		publicKey *sslibsv.SSLibKey
+		err       error
+	)
+	if signerSSH, isSSHSigner := signer.(*ssh.Signer); isSSHSigner {
+		publicKey = signerSSH.MetadataKey()
```

Comment:
need to work this out a bit; I wonder if when we add GPG and Sigstore, we just extend the signer interface in gittuf to include MetadataKey. Right now, the else is only reachable in tests because of the legacy format if I'm not mistaken. We can't hit it from the CLI with this PR.

Reply:
I do recommend to add the public key to the `Signer` interface. See https://github.com/secure-systems-lab/securesystemslib/issues/605

Reply:
`Public()` itself is on the Signer since it's on the Verifier now. The question is whether we expect it to return crypto.PublicKey or the metadata-specific format I've got with MetadataKey()?

Reply:
IMO we don't need `Public()`, if `MetadataKey()` is on the interface:

https://github.com/gittuf/gittuf/blob/e246f934a81279512b3676814d0bae8780af93d2/internal/signerverifier/ssh/ssh.go#L54-L55

Reply:
let's resolve this upstream separately?